akka-http-core/src/main/scala/akka/http/scaladsl/model/headers/HttpChallenge.scala

```diff
@@ -11,9 +11,9 @@ import akka.http.impl.util.JavaMapping.Implicits._
 
 final case class HttpChallenge(scheme: String, realm: String,
                                params: Map[String, String] = Map.empty) extends jm.headers.HttpChallenge with ValueRenderable {
-
   def render[R <: Rendering](r: R): r.type = {
-    r ~~ scheme ~~ " realm=" ~~#! realm
+    r ~~ scheme
+    if (realm != null && !realm.isEmpty) r ~~ " realm=" ~~#! realm
```

Comment:
Not sure about the `isEmpty` check. If in empty string was provided (instead of `null`/`None`) then maybe the user *really* wanted to get it rendered?